src/test/java/org/jitsi/meet/test/web/WebTestBase.java

```diff
@@ -278,35 +278,43 @@ private WebParticipant joinParticipant(
         JitsiMeetUrl            meetURL,
         ParticipantOptions      options)
     {
-        WebParticipant p = participants.get(index);
-
-        if (p == null)
+        WebParticipant p;
+        if (index > -1)
         {
-            // There's an assumption that the participants are created
-            // starting from 0, 1, 2, so throw an Exception if they happen to be
-            // created in different order.
-            int size = participants.getAll().size();
-            if (index != size)
+            p = participants.get(index);
+
+            if (p == null)
             {
-                throw new IllegalArgumentException(
-                    String.format(
-                        "New participant would have been inserted at different "
-                            + "index than expected. Index: %d, size %d.",
-                        index,
-                        size));
+                // There's an assumption that the participants are created
+                // starting from 0, 1, 2, so throw an Exception if they happen to be
+                // created in different order.
+                int size = participants.getAll().size();
+                if (index != size)
+                {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            "New participant would have been inserted at different "
+                                + "index than expected. Index: %d, size %d.",
+                            index,
+                            size));
+                }
+
+                String configPrefix = "web.participant" + (index + 1);
+
+                p = participants.createParticipant(configPrefix, options);
             }
-
-            String configPrefix = "web.participant" + (index + 1);
-
-            p = participants.createParticipant(configPrefix, options);
-
-            // Adds a print in the console/selenium-node logs
-            // useful when checking crashes or failures in node logs
-            p.executeScript(
-                    "console.log('--- Will start test:"
-                        + getClass().getSimpleName() + "')");
+        }
+        else
+        {
+            p = participants.createParticipant("", options);
```

Comment:
what happens here?

Reply:
why call create participant again if it exists?

Reply:
@paweldomas this else clause belongs to the outer if statement that checks the value of the index. This branch executes when `index < 0` which is what `joinNextParticipant` passes down to this method. We use this newly introduced method in malleus where we don't care about what index a participant is placed at, but we need to create arbitrarily many participants.

Reply:
Just leaving this as a note here that we have discussed with George that he will make an attempt to auto select the next index instead of allowing the case with empty prefix.